pkg/persistence/disk_persistence_test.go

```diff
@@ -46,8 +51,30 @@ func TestDiskPersistence_Save(t *testing.T) {
 	}
 }
 
+func TestDiskPersistence_StoragePermission(t *testing.T) {
+	tempDir := "./data_storage"
+
+	err := os.Mkdir(tempDir, 0000) // d---------
+	if _, err := os.Stat(tempDir); os.IsNotExist(err) {
+		t.Fatalf("dir [%+v] was supposed to be created", tempDir)
+	}
+	defer os.RemoveAll(tempDir)
+
+	_, err = NewDiskHandle(tempDir)
+	if !strings.Contains(err.Error(), errExpectedRead.Error()) {
```

Comment:
Please use `reflect.DeepEqual` directly on `err` and `errExpectedRead` because now `err` still can be `nil`.

Reply:
I we don't use `reflect.DeepEqual` we can do `err == nill || !strings.Contains(err.Error(), errExpectedRead.Error())`

Reply:
yea, I'll add a check for nil, regarding `reflect.DeepEqual` see [this](https://github.com/keep-network/keep-common/pull/25#discussion_r406120240) comment